operations/security_and_authentication/Audit_Logs.md

```diff
@@ -0,0 +1,110 @@
+       To enable or disable audit logging, use the following flags with the `csi config init` command. For more information on using flags, see `csi config init -h`:
+
+       `ncn-mgmt-node-auditing-enabled`: Set to `true` to enable host logging or to `false` to disable host logging.
+
+       ```console
+       ncn# csi config init --ncn-mgmt-node-auditing-enabled=true [other config init options]
+       ```
+
+       `k8s-api-auditing-enabled`: Set to `true` to enable Kubernetes API logging or to `false` to disable Kubernetes API logging.
+
+       ```console
+       ncn# csi config init --k8s-api-auditing-enabled=true [other config init options]
+       ```
+
+     * **Option 2**
+
+       Adjust the audit log settings by editing the `system_config.yaml` file.
+
+       View the current settings with the following command:
+
+       ```bash
+       ncn# cd /var/www/ephemeral/prep
+       ncn# grep audit system_config.yaml
+       ```
+
+       Example output:
+
+       ```text
+       k8s-api-auditing-enabled: false
+       ncn-mgmt-node-auditing-enabled: false
+       ```
+
+   * Post-install
+
+     If a system is installed, audit logging can be enabled through the `csi` command:
+
+     * `ncn-mgmt-node-auditing-enabled`
+
+       ```bash
+       ncn-m# export TOKEN=$(curl -k -s -S -d grant_type=client_credentials \
+       -d client_id=admin-client \
+       -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` \
+       https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+       ncn-m# csi handoff bss-update-param --limit <mgmt-node-xname> --set ncn-mgmt-node-auditing-enabled=true
+       ```
+
+     * `k8s-api-auditing-enabled`
+
+       ```bash
+       ncn-m# export TOKEN=$(curl -k -s -S -d grant_type=client_credentials \
+       -d client_id=admin-client \
+       -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` \
+       https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+       ncn-m# csi handoff bss-update-param --limit <mgmt-node-xname> --set k8s-api-auditing-enabled=true
+       ```
+
+     **NOTE**: If `csi` command is not installed, locate the `cray-site-init-*` rpm on the NCN master node and install it:
+
+     ```bash
+     ncn-m# cd /mnt/pitdata
+     ncn-m# find . -name cray-site-init*
+     ncn-m# rpm -Uvh --force <rpm file path>
+     ```
+
+2. Verify if audit logging is enabled (changes made post-install won't be reflected until after the NCN is rebooted)
+
+   * `ncn-mgmt-node-auditing-enabled`
+  
+     ```bash
+     ncn# craysys metadata get ncn-mgmt-node-auditing-enabled
+     ```
+
+   * `k8s-api-auditing-enabled`
+
+     ```bash
+     ncn# craysys metadata get k8s-api-auditing-enabled
+     ```
+
+3. Restart each NCN to apply the new settings after the CSI setting is changed.
```

Comment:
If this is being done during an install and ncn-m001 is still the PIT node, will this automatically get picked up when it is redeployed as ncn-m001?

Reply:
@bo-quan @bklei can either of you answer Mitch's question above?

Reply:
i thought if you are fresh install, because we did something in csi, all nodes will have this enabled? i think reboot is only needed for a system that was installed without audit enabled?

Reply:
@dzou-hpe would it be sufficient to state this at the beginning of the step then?

> Skip this step if the system was installed with audit logging enabled.

Reply:
@nrockershousen I agree with @dzou-hpe -- if this is a fresh install case, setting this should cause the NCNs to come up initially with auditing enabled, so the restart would only be needed if applying after the fact.

Reply:
Thanks! Fixed.

Reply:
Well, wait, is that accurate? If someone follows this procedure on the PIT node during an install, it's possible they'll be doing it after all of the NCNs have already been booted (just not before the PIT node has been redeployed). I don't see any indication earlier on this page saying that if this is done during a CSM install, it needs to be done before the NCNs are first deployed. So if this procedure IS done after the NCNs are deployed but before the PIT node is redeployed, then wouldn't the nodes still need to be rebooted?

I think it probably is more accurate to just explain that these changes take effect on an NCN at boot time, so any NCNs which are currently booted (other than the PIT node) will not have logging enabled until the next time they are booted. That covers the case where the NCNs aren't booted yet and the case where they are, I think. Does that make sense?

Reply:
Actually, I now see that in the CSM install case, we're making the changes either to the "csi config init" arguments or to the files that "csi config init" generates. So I'm assuming that those steps are intended to be done at that point in the install -- when csi config init is being run. So the real issue is probably that we need to be more explicit up above about when they have to do this during a CSM install. I'll mark this thread as resolved and add a new comment to that part of the document.